### ISSUE TITLE
Delete resources (datasets, users, annotations)

### ISSUE THREAD

**Issue by normanrz** [scalableminds/webknossos#4668]
Currently, there is no way to delete datasets from webKnossos. Similarly, there are only ways to archive/deactivate annotations and users. I think this creates confusion for new users.

Maybe we could rename archive/deactivate to a delete action. There could still be a trash from which the data can be recovered indefinitely, but it might be clearer wording.

For expensive resources such as datasets, we could have a delete action that moves the data into a trash folder which auto-clears after x days. That would make datasets easily recoverable in case of error, but eventually remove them entirely. Removed datasets could be represented as tombstones ⚰️  in the database to not break references.

**Comment by normanrz** [comment]
- [ ] Users 
- [ ] Annotations 
- [x] Datasets #4696

**Comment by philippotto** [comment]
> For expensive resources such as datasets, we could have a delete action that moves the data into a trash folder which auto-clears after x days.

Not sure whether this auto-clearing was already implemented, but I'd find it more safe to only remove datasets which have been more than X days in the trash. If this information is not available via filesystem metadata, we might want to store it ourselves?
Not super critical, probably, but something we should keep in mind.

**Comment by normanrz** [comment]
I don't think that is implemented yet. But what you say is what I actually meant :D
I think this could be implemented outside of wk as a cronjob.
